client/object_put.go

```diff
@@ -222,6 +232,11 @@ func (x *DefaultObjectWriter) Close() error {
 
 	defer x.cancelCtxStream()
 
+	if x.streamClosed {
+		err = x.err
+		return x.closeErr
```

Comment:
`return nil`

Reply:
so closing stream error is ignored _always_, right?

Reply:
We're closing the connection, do we have any issue with closing it? No -> no error. If there was some error _before_ the connection was closed, it had been returned to the user already.

Reply:
I tend to agree with `return nil`, because you got `EOF` during the object uploading and you in the middle of this process. You have just one way - to close the connection. Even it is an error, it doesn't matter - previous one is more important

Reply:
> do we have any issue with closing it?

nobody can say for sure, that is an external library that returns an `error`

Reply:
updated the code, look, please and suggest your ideas